core/sail/api/src/main/java/org/eclipse/rdf4j/sail/helpers/AbstractNotifyingSailConnection.java

```diff
@@ -33,37 +33,28 @@ public AbstractNotifyingSailConnection(AbstractSail sailBase) {
 
 	@Override
 	public void addConnectionListener(SailConnectionListener listener) {
-		synchronized (listeners) {
-			listeners.add(listener);
-		}
+		listeners.add(listener);
```

Comment:
Note that we are already using a thread-safe collection: `private final List<SailConnectionListener> listeners = new CopyOnWriteArrayList<>();`